libraries/AP_Proximity/AP_Proximity_Backend.cpp

```diff
@@ -351,40 +330,32 @@ bool AP_Proximity_Backend::get_next_ignore_start_or_end(uint8_t start_or_end, in
     return found;
 }
 
-void AP_Proximity_Backend::update_locations()
+// returns true if database is ready to be pushed to and all cached data is ready
+bool AP_Proximity_Backend::database_prepare_for_push()
```

Comment:
I'm struggling to see the value of having this spun out like this. Why not just do it inside the push as an early return? Seems more complicated.

Reply:
reworked, thanks for the suggetsion